separate feature flag for Data Registry reports

snopoke opened the pull request:
## Product Description
Splitting out custom web report creation from the other features allows more flexible control over their usage.

## Technical Summary
New FF and one additional check in the view.

## Feature Flag
Data Registry, Data Registry UCR

## Safety Assurance

### Safety story
New FF is used in the same way as the previous one. The only change is an additional validation check in the report view which follows the same pattern as other existing checks.

### Automated test coverage
None

### QA Plan
None


### Migrations
NA

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

snopoke committed:
new toggle

snopoke committed:
update docs link

snopoke committed:
more toggle to put it with related ones

snopoke committed:
use new toggle for data registry UCR creation

snopoke committed:
check access in view

orangejenny commented on corehq/apps/userreports/views.py:
```diff
@@ -509,6 +510,10 @@ def dispatch(self, request, *args, **kwargs):
             self.app_id = self.request.GET.get('application', None)
             self.source_id = self.request.GET['source']
             if self.registry_slug:
+                if not toggles.DATA_REGISTRY_UCR.enabled(self.domain):
```
This check means domains without the flag won't be able to edit data registry UCRs either, right? (which I assume is what we want)

snopoke committed:
perform check for both new and edit reports

snopoke committed:
remove delete button on error page for DR reports

snopoke committed:
don't show edit button on DR report if editing is not enabled

snopoke committed:
include DR doc types for undelete opterations

snopoke commented:
@orangejenny I've updated the PR to more the check. I've also removed the 'delete' button that was on the error screen as well as the 'edit' button on the report view (in both cases only if editing / delete is not permitted).

orangejenny commented on corehq/apps/userreports/views.py:
```diff
@@ -535,12 +536,15 @@ def dispatch(self, request, *args, **kwargs):
+            }
         else:
             context = {}
+        if allow_delete is not None:
```
Nitpick: this code would behave the same without the if statement, wouldn't it?

snopoke replied:
no, without the `if` it would always put the argument value into the context which would override any value already in the context.
